presto-main/src/main/java/com/facebook/presto/execution/resourceGroups/ResourceGroupManager.java

```diff
@@ -25,4 +29,6 @@
     void addConfigurationManagerFactory(ResourceGroupConfigurationManagerFactory factory);
 
     void loadConfigurationManager() throws Exception;
+
+    List<ResourceGroup> getRootResourceGroups();
```

Comment:
Although the `ResourceGroup ` interface doesn't have the notion of a root, the `ResourceGroupId` class (that you can get with `ResourceGroup::getId()`) has a `getParent()` method, which indicates there's a hierarchy/structure to the resource groups -- `InternalResourceGroupManager` uses `getParent()` to see whether a group is root or not.

Another option is to introduce `ResourceGroupManager::getResourceGroups()` instead of `ResourceGroupManager::getRootResourceGroups()` that return all the groups, and we let the caller figure out the hierarchy.